draft-nygren-httpbis-httpssvc.md

```diff
@@ -0,0 +1,791 @@
+to determine if an HTTPSSVC record is available for that origin.  To do so,
+the client SHOULD construct a corresponding "https" URL as follows:
+
+1. Replace the "http" scheme with "https".
+
+2. If the "http" URL explicitly specifies port 80, specify port 443.
+
+3. Do not alter any other aspect of the URL.
+
+This construction is equivalent to {{HSTS}} Section 8.3, point 5.
+
+If an HTTPSSVC record is present for this "https" URL, the client
+should treat this as the equivalent of receiving an HTTP "302 Found"
+redirect to the "https" URL.
+Because HTTPSSVC is received over an often insecure channel (DNS),
+clients MUST NOT place any more trust in this signal than if they
+had received a 302 redirect over cleartext HTTP.
+
+
+## Cache interaction
+
+If the client has an Alt-Svc cache, and a usable Alt-Svc value is
+present in that cache, then the client SHOULD NOT issue an HTTPSSVC DNS
+query.  Instead, the client SHOULD proceed with alternative service
+connection as usual.
+
+If the client has a cached Alt-Svc entry that is expiring, the
+client MAY perform an HTTPSSVC query to refresh the entry.
+
+## Optimizing for performance
+
+Clients that are optimizing for performance (i.e. minimum connection
+setup time) SHOULD implement the following connection sequence:
+
+1. Issue address (AAAA and/or A) queries, immediately followed by the
+   HTTPSSVC query.
+1. If an HTTPSSVC response is received first, proceed with alternative
+   service connection and ignore the address responses if they are no
+   longer relevant.
+1. Otherwise, initiate connection to the origin server.
+1. As soon as an Alt-Svc field value is received, through the DNS or
+   over HTTP, proceed with alternative service connection.  Do not
+   abort this connection if an Alt-Svc field value is received from the
+   other source later.
+
+If the HTTPSSVC and address queries return approximately simultaneously,
+this process typically saves three roundtrips on a fresh connection
+that uses Alt-Svc: one each for TCP, TLS 1.3, and HTTP.  (On subsequent
+connections, the Alt-Svc information is expected to be cached, so this
+procedure does not apply.)
+
+If a client can cache Alt-Svc entries that were received over both HTTP
+and DNS, the client MAY prefer entries that were received over HTTP.
+These records may be more narrowly targeted for the specific client.
+
+As an additional optimization, when choosing among multiple Alt-Svc
+values, clients MAY prefer those that will not require an address
+query, either because the corresponding address record is
+already in cache or because the host is an IP address.
+
+Note that this procedure does not rely on recursive resolvers handling
+the HTTPSSVC record type correctly.  If HTTPSSVC queries receive spurious
+NXDOMAIN responses, or even no response at all, connections will proceed
+as usual without any delay.
+
+## Optimizing for privacy
+
+Clients that are optimizing for privacy SHOULD implement {{!AltSvcSNI}}
```

Comment:
This references the draft which defines sni=, but you redefine it in 6.1.  Need to resolve.